src/main/plugins/org.dita.pdf2/xsl/fo/root-processing.xsl

```diff
@@ -265,9 +265,21 @@ See the accompanying license.txt file for applicable licenses.
     <xsl:apply-templates select="*" mode="generatePageSequences"/>
   </xsl:template>
   <xsl:template match="*[contains(@class, ' map/topicref ')]" mode="generatePageSequences" priority="0">
-    <xsl:for-each select="key('topic-id', @id)">
-      <xsl:call-template name="processTopicSimple"/>
-    </xsl:for-each>
+    <xsl:choose>
+      <xsl:when test="ancestor::*[contains(@class,' bookmap/frontmatter ')]">
+        <!-- To fit the pattern, this should be in its own match template. But a general match for frontmatter/*
+             conflicts with priority of existing rules (e.g., preface); changing priorities would
+             break customizations. --> 
+        <xsl:for-each select="key('topic-id', @id)">
+          <xsl:call-template name="processFrontMatterTopic"/>
+        </xsl:for-each>
+      </xsl:when>
+    <xsl:otherwise>

```

Comment:
Duh. Coding eyes must be tired right now. I'll fix these and all the other notes, thanks.